web_external/ImagesGallery/ImagesView.js

```diff
@@ -0,0 +1,128 @@
+isic.views.ImagesView = isic.View.extend({
+    initialize: function (settings) {
+        this.completeFacets = new isic.collections.ImagesFacetCollection();
+        // TODO: when filteredFacets fetch returns no images, all models are gone / removed
+        this.filteredFacets = new isic.collections.ImagesFacetCollection();
+        this.filters = new isic.collections.ImagesFilters();
+
+        // TODO: replace this with an inline grammar
+        this.filterLoaded = isic.SerializeFilterHelpers.loadFilterGrammar();
+
+        this.images = new isic.collections.SelectableImageCollection();
+        this.images.pageLimit = 50;
+
+        // Rather than issue a second fetch request for this.filteredFacets,
+        // copy the response of the first request for this.completeFacets
+        this.listenTo(this.completeFacets, 'sync', function (collection, resp, options) {
+            // TODO: ideally, this.filteredFacets would only be reset when
+            // this.completeFacets triggered an "update" (which would prevent resets when
+            // "sync" had no changes); however, it's difficult to deep-copy all of
+            // the models from this.completeFacets to this.filteredFacets without having
+            // access to the raw HTTP response
+            this.filteredFacets.reset(resp, {parse: true});
+
+            this.images.fetch();
+
+            this.filters.initialize(collection);
+        });
+        this.listenTo(this.filters, 'change', function () {
+            // TODO: Ideally, this should select and cancel only the old set of requests; until
+            // that's fixed, hopefully nobody will click histogram buttons too early in the page
+            // load
+            girder.cancelRestRequests();
```

Comment:
Replace this with a mechanism that saves the two requests made (a few lines below) and only cancel those. This mechanism can break the application because there may be other pending requests that should *not* be canceled by action here.